npm/ipsm/ipsm.go

```diff
@@ -267,11 +291,19 @@ func (ipsMgr *IpsetManager) AddToSet(setName, ip, spec, podUid string) error {
 		return nil
 	}
 
-	if !ipsMgr.SetExists(setName, spec) {
+	if ip == "" {
```

Comment:
This alone check may not work, since in case of hash:ip,port, the ip will be "1.1.1.1,tcp:80", but if the Ip is empty then at this stage we will get ",tcp:80"